### PR TITLE
Revert fmt.sh and update make format to use fmt.sh

### PR DESCRIPTION
**Pull request by xiaolanz**
consolidate style using fmt.sh

### CODE REVIEW

**Comment by mandarjog** [comment]
/lgtm

As a followup we should ensure that linter uses the same mechanism for enforcement.

**Comment by xiaolanz** [comment]
linter seems fine.

File: lintconfig_base.json [hunk 1]
```diff
@@ -9,7 +9,7 @@
     "test": true,
     "sort": ["path"],
     "Linters": {
-        "goimports": {"Command": "goimports -l --local istio.io"}
```
Comment (xiaolanz) [inline lintconfig_base.json]: @mandarjog I have problem changing this. The linter stops to work deterministically.

Comment (mandarjog) [reply lintconfig_base.json]: `bin/fmt.sh` does not work ?

**Comment by mattdelco** [comment]
It seems like this could be done in the Makefile using something like:
<pre>
GOIMPORTS := $(shell which goimports 2>/dev/null || echo "${ISTIO_BIN}/goimports" )
${ISTIO_BIN}/goimports: | ${ISTIO_BIN}
	unset GOOS && unset GOARCH && go get golang.org/x/tools/cmd/goimports

GO_FILES_CMD := find . -type f -name '*.go' ! -name '*.gen.go' ! -name '*.pb.go' ! -name '*mock*.go' | grep -v .vendor

ifeq ($(shell uname),Darwin)
   SED_OPT := ''
endif

# eval is used to compute file list on demand (rather than on every invocation of make)
format: | $(GOIMPORTS)
	$(eval GO_FILES:=$(shell $(GO_FILES_CMD)))
	for fl in $(GO_FILES); do sed -i $(SED_OPT) -e "/^import[[:space:]]*(/,/)/{ /^\s*$$/d;}" $$fl; done && \
	gofmt -s -w $(GO_FILES) && \
	$(GOIMPORTS) -w -local istio.io $(GO_FILES)
</pre>

Otherwise, the existing GO_FILES_CMD variable and 2 adjoining variables in the Makefile can probably be removed since afaict they'd no longer be used anywhere.

File: lintconfig_base.json (hunk 1 above)
Comment (xiaolanz) [reply lintconfig_base.json]: Found the problem. I should create a check only model for fmt.sh.

**Comment by xiaolanz** [comment]
moved to https://github.com/istio/istio/pull/4785

**Pull request closed without merging**